### PR TITLE
E2e tests

### PR DESCRIPTION
**Pull request by ofrogon**
- Add new e2e test.
- Activate the ignored unit tests.
- Fix error when calling the parameters `--deploy` and `--version` where the current working directory will not be the one specified after `--deploy`.
- Fix the HTTP header information when pushing the package to the server.

### CODE REVIEW

**Comment by ofrogon** [comment]
@franknarf8 This is still waiting your code review...

File: lib/utils.js [hunk 1]
```diff
@@ -206,7 +206,13 @@ function getBowerContent(from) {
...
  */
 function getNpmRc(callback) {
-    let npmRcFilePath = path.join(os.homedir(), ".npmrc");
+    let npmRcFilePath;
+
+    if (process.env.NODE_ENV !== "test") {
+        npmRcFilePath = path.join(os.homedir(), ".npmrc");
+    } else {
+        npmRcFilePath = path.join(__dirname, "..", "test/data/.npmrc");
+    }
```
Comment (franknarf8) [inline lib/utils.js]: Embed test code in regular code is not recommended.

Comment (franknarf8) [inline lib/utils.js]: If you would encapsulate that in an object, you could have a variable `npmrcfinder` that you could dependency inject with different piece of code in test and in production.

Comment (ofrogon) [inline lib/utils.js]: OK, since this Pull Request is one month old, I will make it pass and log your comment in an enhancement issue @franknarf8 .

**Pull request merged**